### PR TITLE
PAL cleanup

### PR DESCRIPTION
**Pull request by kangaroo**
This PR removes dead solaris and powerpc code.  It also removes every configure check that is not used in the repository.  Finally, there was an inconsistency in the source and configure stage between HAVE_FTRUNCATE_LENGTH_BUG and HAVE_FTRUNCATE_LENGTH_ISSUE.  This is addressed, needing one minor change in file.cpp.

### CODE REVIEW

File: src/pal/src/file/file.cpp [hunk 1]
```diff
@@ -2505,6 +2482,7 @@ CorUnix::InternalSetEndOfFile(
     // extend the file to consume the remainder of free space.
     // 
     struct statfs sFileSystemStats;
+    off_t cbFreeSpace;
```
Comment (janvorli) [inline src/pal/src/file/file.cpp]: Why have you moved the declaration herehere? We prefer variable declaration with initialization unless there is a reason for separating them.

**Comment by janvorli** [comment]
LGTM modulo my comment.

File: src/pal/src/file/file.cpp (hunk 1 above)
Comment (kangaroo) [inline src/pal/src/file/file.cpp]: @janvorli clang required it.  The decl created a protected scope, and the goto violated that.

Comment (janvorli) [inline src/pal/src/file/file.cpp]: @kangaroo Ah, I've missed the goto. Then it makes sense.

**Comment by sergiy-k** [comment]
LGTM

**Pull request merged**